index.js

```diff
@@ -1448,14 +1447,179 @@ module.exports = {
+    }
+
+    // calculate the cost (in ether) of this transaction
+    // (note: this is an *upper bound* on the cost, set by the gasLimit)
+    var cost = etx.getUpfrontCost().toString();
+
+    // validate the transaction's signature
+    if (!etx.validate()) return callback(errors.TRANSACTION_INVALID);
+
+    // submit the raw transaction to the network
+    if (!isFunction(callback)) {
+      var rawTransactionResponse = this.sendRawTransaction(etx.serialize().toString("hex"));
+      return this.parseRawTransactionResponse(rawTransactionResponse, packaged, address, privateKey, cost);
+    }
+    var self = this;
+    this.sendRawTransaction(etx.serialize().toString("hex"), function (rawTransactionResponse) {
+      self.parseRawTransactionResponse(rawTransactionResponse, packaged, address, privateKey, cost, callback);
+    });
+  },
+
+  // set transaction nonce to the number of transactions
+  setNonceThenSubmitRawTransaction: function (packaged, address, privateKey, callback) {
+    var self = this;
+    if (packaged.nonce) {
+      return this.submitRawTransaction(packaged, address, privateKey, callback);
+    }
+    if (!isFunction(callback)) {
+      var txCount = this.pendingTxCount(address);
+      if (this.debug.nonce) {
+        console.debug('[ethrpc] txCount:', parseInt(txCount, 16));
+      }
+      if (txCount && !txCount.error && !(txCount instanceof Error)) {
+        packaged.nonce = abi.hex(txCount);
+      }
+      return this.submitRawTransaction(packaged, address, privateKey);
+    }
+    this.pendingTxCount(address, function (txCount) {
+      if (self.debug.nonce) {
+        console.debug('[ethrpc] txCount:', parseInt(txCount, 16));
+      }
+      if (txCount && !txCount.error && !(txCount instanceof Error)) {
+        packaged.nonce = abi.hex(txCount);
+      }
+      self.submitRawTransaction(packaged, address, privateKey, callback);
+    });
+  },
+
+  packageAndSubmitRawTransaction: function (payload, address, privateKey, callback) {
+    var self = this;
+    if (!payload || payload.constructor !== Object) {
+      if (!isFunction(callback)) throw new this.Error(errors.TRANSACTION_FAILED);
+      return callback(errors.TRANSACTION_FAILED);
+    }
+    if (!address || !privateKey) {
+      if (!isFunction(callback)) throw new this.Error(errors.NOT_LOGGED_IN);
+      return callback(errors.NOT_LOGGED_IN);
+    }
+
+    // parse and serialize transaction parameters
+    var packaged = this.packageRequest(payload);
+    packaged.from = address;
+    packaged.nonce = payload.nonce || "0x0";
+    packaged.value = payload.value || "0x0";
+    if (payload.gasLimit) {
+      packaged.gasLimit = abi.hex(payload.gasLimit);
+    } else if (this.block && this.block.gasLimit) {
+      packaged.gasLimit = abi.hex(this.block.gasLimit);
+    } else {
+      packaged.gasLimit = this.DEFAULT_GAS;
+    }
+    if (this.networkID && !isNaN(parseInt(this.networkID, 10)) && parseInt(this.networkID, 10) < 109) {
```

Comment:
What is the magic number `109` for?  Where does it come from?  Why are we comparing against it?